src/Adapter/MSTest.TestAdapter/MSTestSettings.cs

```diff
@@ -262,32 +264,49 @@ public static void PopulateSettings(MSTestSettings settings)
     /// <param name="logger"> The logger for messages. </param>
     /// The discovery context that contains the runsettings.
     /// </param>
-    internal static void PopulateSettings(IDiscoveryContext? context, IMessageLogger? logger)
+    internal static void PopulateSettings(IDiscoveryContext? context, IMessageLogger? logger, IConfiguration? configuration)
     {
-        RunConfigurationSettings = RunConfigurationSettings.PopulateSettings(context);
-
-        if (context?.RunSettings == null || StringEx.IsNullOrEmpty(context.RunSettings.SettingsXml))
+        if (configuration != null && context?.RunSettings != null)
```

Comment:
We already discussed this one related to the possible `RunConfig` usage by MSTest.